hail_search/queries/base.py

```diff
@@ -1001,7 +1007,7 @@ def _filter_compound_hets(self):
         def key(v):
             ks = [v[k] for k in self.KEY_FIELD]
             return ks[0] if len(self.KEY_FIELD) == 1 else hl.tuple(ks)
-        ch_ht = ch_ht.annotate(key_=key(ch_ht.row), gene_ids=self._gene_ids_expr(ch_ht))
+        ch_ht = ch_ht.annotate(key_=key(ch_ht.row), gene_ids=self._gene_ids_expr(ch_ht, filtered_genes_only=True))
```

Comment:
this change means that if a gene filter has been applied we will only compute possible comp het pairs in those genes